src/analyzer/analyzermanager.cpp

```diff
@@ -0,0 +1,296 @@
+    //being analyzed by the background worker. We cannot reuse the background worker, but we should discard its work.
+    if (!m_prioTrackQueue.contains(tio)) {
+        m_prioTrackQueue.append(tio);
+        if (m_priorityWorkers.size() < m_MaxThreads) {
+            createNewWorker(WorkerType::priorityWorker);
+            if (m_priorityWorkers.size() + m_defaultWorkers.size() > m_MaxThreads) {
+                AnalyzerWorker * backwork = m_defaultWorkers.first();
+                backwork->pause();
+                //Ideally i would have done this on the slotPaused slot, but then i cannot 
+                //ensure i won't call pause twice for the same worker.
+                m_pausedWorkers.append(backwork);
+                m_defaultWorkers.removeAll(backwork);
+            }
+        }
+    }
+}
+// This is called from the GUI for the analysis feature of the library.
+void AnalyzerManager::queueAnalyseTrack(TrackPointer tio) {
+    if (!m_defaultTrackQueue.contains(tio)) {
+        m_defaultTrackQueue.append(tio);
+        if (m_pausedWorkers.size() + m_defaultWorkers.size() < m_MaxThreads) {
+            createNewWorker(WorkerType::defaultWorker);
+        }
+    }
+}
+
+// This slot is called from the decks and samplers when the track is loaded.
+void AnalyzerManager::slotAnalyseTrack(TrackPointer tio) {
+    analyseTrackNow(tio);
+}
+
+
+//slot
+void AnalyzerManager::slotUpdateProgress(int workerIdx, struct AnalyzerWorker::progress_info* progressInfo) {
+    //Updates to wave overview and player status text comes from a signal emited from the track by calling setAnalyzerProgress.
+    progressInfo->current_track->setAnalyzerProgress(progressInfo->track_progress);
+    //These update the Analysis feature and analysis view.
+    emit(trackProgress(workerIdx, progressInfo->track_progress / 10));
+    if (progressInfo->track_progress == 1000) {
+        //Right now no one is listening to trackDone, but it's here just in case.
+        emit(trackDone(progressInfo->current_track));
+        //Report that a track analysis has finished, and how many are still remaining.
+        emit(trackFinished(m_defaultWorkers.size() + m_defaultTrackQueue.size() - 1));
+    }
+    //TODO: Which is the consequence of not calling reset?
+    progressInfo->current_track.reset();
+    progressInfo->sema.release();
+}
+
+void AnalyzerManager::slotNextTrack(AnalyzerWorker* worker) {
+    //TODO: The old scan checked in isLoadedTrackWaiting for pTrack->getAnalyzerProgress()
+    // and either tried to load a previuos scan, or discarded the track if it had already been
+    // analyzed. I don't fully understand the scenario and I am not doing that right now.
+
+    //This is used when the maxThreads change. Extra workers are paused until active workers end.
+    //Then, those are terminated and the paused workers are resumed.
+    TrackPointer track;
+    AnalyzerWorker* forepaused=nullptr;
+    foreach(AnalyzerWorker* worker, m_pausedWorkers) {
```

Comment:
Ouch! 👍 Indeed, i added that when adding the refresh of the threads from the preferences, but I overlooked the names.